File: account/src/main/java/serp/project/account/kernel/property/JwtProperties.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.account.kernel.property;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@ConfigurationProperties(prefix = "app.security.jwt")
@Data
public class JwtProperties {
    private String privateKey;
    private String publicKey;
    private Long accessTokenExpiration;
    private Long refreshTokenExpiration;
    private String header;
    private String prefix;
}

File: ptm_task/src/kernel/properties/jwt_properties.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package properties

import "github.com/golibs-starter/golib/config"

type JwtProperties struct {
	PublicKey string
}

func (j JwtProperties) Prefix() string {
	return "app.security.jwt"
}

func NewJwtProperties(loader config.Loader) (*JwtProperties, error) {
	var props JwtProperties
	if err := loader.Bind(&props); err != nil {
		return nil, err
	}
	return &props, nil
}
